volatility3/framework/plugins/linux/sockstat.py

```
@@ -0,0 +1,506 @@
+        for net in nethead.to_list(net_symname, "list"):
+            net_device_symname = self._vmlinux.symbol_table_name + constants.BANG + "net_device"
+            for net_dev in net.dev_base_head.to_list(net_device_symname, "dev_list"):
+                if net.get_inode() != netns_id:
+                    continue
+                dev_name = utility.array_to_string(net_dev.name)
+                netdevices_map[net_dev.ifindex] = dev_name
+        return netdevices_map
+
+    def process_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str], Dict]:
+        """Takes a kernel generic `sock` object and processes it with its respective socket family
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns a tuple with:
+            sock: The respective kernel's *_sock object for that socket family
+            sock_stat: A tuple with the source, destination and state strings.
+            extended: A dictionary with key/value extended information.
+        """
+        family = sock.get_family()
+        extended = {}
+        sock_handler = self._sock_family_handlers.get(family)
+        if sock_handler:
+            try:
+                unix_sock, sock_stat = sock_handler(sock)
+                self._update_extended_socket_filters_info(sock, extended)
+
+                return unix_sock, sock_stat, extended
+            except exceptions.SymbolError as e:
+                # Cannot finds the *_sock type in the symbols
+                vollog.log(constants.LOGLEVEL_V, "Error processing socket family '%s': %s", family, e)
+        else:
+            vollog.log(constants.LOGLEVEL_V, "Unsupported family '%s'", family)
+
+        # Even if the sock family is not supported, or the required types
+        # are not present in the symbols, we can still show some general
+        # information about the socket that may be helpful.
+        saddr_tag = daddr_tag = "?"
```

Hard coding "we don't know" values into functions is a bad idea.  This should either throw an exception or use an unambiguous value such as `None`, anything consuming the output of this will need to check for bad values.